test/policy.bats

```diff
@@ -52,6 +52,26 @@ SANDBOX_CONFIG="$TESTDATA/sandbox_config.json"
 	assert_log "$RESTRICTIVE_POLICY"
 }
 
+@test "deny unsigned image if already pulled and container created during mounting phase" {
+    start_crio
+    crictl pull "$UNSIGNED_IMAGE"
+    POD_ID=$(crictl runp "$TESTDATA/sandbox_config.json")
+    CTR_CONFIG="$TESTDIR/config.json"
+    jq '.image.image = "'"$UNSIGNED_IMAGE"'" | .image.user_specified_image = "'"$UNSIGNED_IMAGE"'"' "$TESTDATA/container_config.json" > "$CTR_CONFIG"
+    crictl create "$POD_ID" "$CTR_CONFIG" "$TESTDATA/sandbox_config.json"
+    stop_crio_no_clean
+
+    SIGNATURE_POLICY="$RESTRICTIVE_POLICY" start_crio
+    POD_ID=$(crictl runp "$TESTDATA/sandbox_config.json")
+    CTR_CONFIG="$TESTDIR/config.json" 
+    jq '.image.image = "'"$UNSIGNED_IMAGE"'" | .image.user_specified_image = "'"$UNSIGNED_IMAGE"'"' "$TESTDATA/container_config.json" > "$CTR_CONFIG"
+
+    run ! crictl create "$POD_ID" "$CTR_CONFIG" "$TESTDATA/sandbox_config.json"
+
+    [[ "$output" == *"SignatureValidationFailed"* ]]
+    assert_log "$RESTRICTIVE_POLICY"
+}
```

Comment:
You need to set the image in the mounts as well, otherwise the added code path will not be triggered. See:

https://github.com/cri-o/cri-o/blob/e7cdc81463713dd822c2194ea9d2e9b17e31e398/test/oci_volumes.bats#L18-L57